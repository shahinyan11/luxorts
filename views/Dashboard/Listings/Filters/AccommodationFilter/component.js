import { Dropdown, Form } from 'antd';
import { FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';
import { FormikProvider } from 'formik';
import classNames from 'classnames';

import { ACCOMMODATION_VALIDATION, LISTING_FILTER_KEY } from 'constants/listing/manageListings';

import SvgIcon from 'views/shared/SvgIcon';
import GradientButton from 'views/shared/GradientButton';
import InputNumberField from 'views/shared/form/InputNumberField';

import useContainer from './hook';

const AccommodationFilter = ({ filters, onFilter }) => {
  const {
    visible,
    clearButtonVisible,
    formik,
    appliedFilters,
    onVisibilityChangeHandler,
    onApply,
    onClear,
  } = useContainer({
    onFilter,
    filters,
  });
  const hasFilters = appliedFilters > 0;

  return (
    <Dropdown
      trigger="click"
      visible={visible}
      onVisibleChange={onVisibilityChangeHandler}
      placement="bottomLeft"
      className="table-head__dropdown-payment"
      overlay={
        <div className="table-head__dropdown-content">
          <div className="d-flex flex-column mb-0 table-head__content">
            <Form layout="vertical" size="large" className="pl-24 pr-24 pt-24 pb-24">
              <FormikProvider value={formik}>
                <InputNumberField
                  name={LISTING_FILTER_KEY.GUESTS_NUMBER_EQ}
                  formItemClassName="mb-8 width-312 ant-form-item--light"
                  addonBefore={<FormattedMessage id="shared.guests" />}
                  min={ACCOMMODATION_VALIDATION.MIN}
                  max={ACCOMMODATION_VALIDATION.MAX}
                />
                <InputNumberField
                  name={LISTING_FILTER_KEY.BEDROOMS_AMOUNT_EQ}
                  formItemClassName="mb-8 width-312 ant-form-item--light"
                  addonBefore={<FormattedMessage id="shared.bedrooms" />}
                  min={ACCOMMODATION_VALIDATION.MIN}
                  max={ACCOMMODATION_VALIDATION.MAX}
                />
                <InputNumberField
                  name={LISTING_FILTER_KEY.BEDS_AMOUNT_EQ}
                  formItemClassName="mb-8 width-312 ant-form-item--light"
                  addonBefore={<FormattedMessage id="shared.beds" />}
                  min={ACCOMMODATION_VALIDATION.MIN}
                  max={ACCOMMODATION_VALIDATION.MAX}
                />
                <InputNumberField
                  name={LISTING_FILTER_KEY.BATHROOMS_AMOUNT_EQ}
                  formItemClassName="mb-0 width-312 ant-form-item--light"
                  addonBefore={<FormattedMessage id="shared.baths" />}
                  min={ACCOMMODATION_VALIDATION.MIN}
                  max={ACCOMMODATION_VALIDATION.MAX}
                />
              </FormikProvider>
            </Form>
          </div>
          <div className="table-head__actions pt-24 pb-24 d-flex justify-content-flex-end">
            {clearButtonVisible && (
              <GradientButton
                onClick={onClear}
                className="main-btn--medium mr-16"
                variant="tertiary"
                text={{ id: 'shared.clear' }}
              />
            )}
            <GradientButton
              onClick={onApply}
              className="main-btn--medium min-width-120"
              text={{ id: 'shared.apply' }}
            />
          </div>
        </div>
      }
    >
      <GradientButton
        className={classNames('main-btn--medium', hasFilters && 'main-btn--filter-active')}
        variant="filter"
        text={
          hasFilters
            ? {
                id: 'shared.accommodationCount',
                values: { count: <span className="main-btn__amount">{appliedFilters}</span> },
              }
            : { id: 'shared.accommodation' }
        }
        addonAfter={<SvgIcon icon="arrow-down" className="icon-right" />}
      />
    </Dropdown>
  );
};

AccommodationFilter.propTypes = {
  filters: PropTypes.shape().isRequired,
  onFilter: PropTypes.func.isRequired,
};

export default AccommodationFilter;
