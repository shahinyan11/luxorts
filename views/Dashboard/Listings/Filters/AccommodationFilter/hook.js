import { useCallback, useMemo } from 'react';
import { pick } from 'ramda';
import { useFormik } from 'formik';

import {
  ACCOMMODATION_LISTING_FILTER_KEYS,
  LISTING_FILTERS,
} from 'constants/listing/manageListings';

import useImmutableCallback from 'hooks/useImmutableCallback';
import useAntdDropdownVisible from 'hooks/useAntdDropdownVisible';

import validationSchema from 'lib/yupLocalised/scheme/manageListings/accommodationFilter';

function useContainer({ filters, onFilter }) {
  const [visible, onVisibilityChangeHandler, setVisible] = useAntdDropdownVisible();

  /**
   * Formik initialization
   */
  const formik = useFormik({
    enableReinitialize: true,
    initialValues: {
      ...filters,
    },
    validationSchema,
  });

  /**
   * Applied filters count
   * @type {number}
   */
  const appliedFilters = useMemo(() => {
    let count = 0;

    Object.values(filters).forEach((value) => {
      if (value) {
        count += 1;
      }
    });

    return count;
  }, [filters]);

  /**
   * Is clear button visible
   * @type {boolean}
   */
  const clearButtonVisible = useMemo(() => {
    let count = 0;

    Object.values(formik.values).forEach((value) => {
      if (value) {
        count += 1;
      }
    });

    return count > 0;
  }, [formik.values]);

  /**
   * Handle apply button click
   * @type {(function(): void)|*}
   */
  const onApply = useCallback(() => {
    if (!formik.isValid) {
      return;
    }

    onFilter(formik.values);
    setVisible(false);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [formik.isValid, formik.values]);

  /**
   * Handle clear button click
   */
  const onClear = useImmutableCallback(() => {
    onFilter(pick(ACCOMMODATION_LISTING_FILTER_KEYS, LISTING_FILTERS));
    formik.resetForm();
  });

  return {
    visible,
    clearButtonVisible,
    formik,
    appliedFilters,
    onVisibilityChangeHandler,
    onApply,
    onClear,
    setVisible,
  };
}

export default useContainer;
