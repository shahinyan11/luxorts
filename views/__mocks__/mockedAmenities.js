export default {
  'Amina Heaney': {
    '5e066ad7-4194-4bd4-95ec-0d9715f8159b': {
      id: '5e066ad7-4194-4bd4-95ec-0d9715f8159b',
      description: 'Maiores porro nostrum dolorum sunt.',
      name: 'Nolan Kertzmann',
      status: 'active',
      category: null,
      checked: false,
    },
  },
  'Torrie Sawayn': {
    'c277033f-1683-4d2d-a409-d00b0bbe8201': {
      id: 'c277033f-1683-4d2d-a409-d00b0bbe8201',
      description: 'Nemo magnam dolores quam voluptatibus distinctio unde.',
      name: 'Stephine Rohan',
      status: 'active',
      category: null,
      checked: false,
    },
  },
};
