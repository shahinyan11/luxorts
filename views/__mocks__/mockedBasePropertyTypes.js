export default [
  {
    value: '51e35121-c27e-46fc-9feb-b51dab2490ee',
    label: 'repudiandae',
    description: 'description repudiandae',
  },
  {
    value: '995c5714-185d-4085-8512-690903334f6e',
    label: 'molestias',
    description: 'description molestias',
  },
];
