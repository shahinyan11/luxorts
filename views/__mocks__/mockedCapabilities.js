export default {
  APPROVED: {
    '11da7909-9f4c-45d6-a2a0-8a105b067ed2': {
      id: '11da7909-9f4c-45d6-a2a0-8a105b067ed2',
      description: 'Vel commodi ad incidunt repellendus a molestias delectus nam.',
      name: 'Tonya Wunsch',
      status: 'active',
      checked: false,
    },
    'b84c898e-2176-4ac8-9dfa-54186ccef6cb': {
      id: 'b84c898e-2176-4ac8-9dfa-54186ccef6cb',
      description: 'Maxime nostrum ex iste velit nulla incidunt.',
      name: 'Edwin Greenholt',
      status: 'active',
      checked: false,
    },
  },
};
