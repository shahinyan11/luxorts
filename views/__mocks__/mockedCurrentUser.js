export default {
  id: '5c5d814c-8e06-433d-b006-4119311585bd',
  email: 'evhenii.hryhoriev+10@rubygarage.org',
  publicId: 'c4toxHVi',
  verifiedStatus: 'unverified',
  userType: 'traveller_account',
  userProfile: {
    id: '52080e19-c5e0-45bb-af58-edbbe70e78fa',
    about: null,
    address: null,
    avatarUrls: {
      original: '/avatar-img-url',
    },
    dateOfBirth: '1994-01-01',
    drivingLicenseFirstName: 'Evhenii',
    drivingLicenseLastName: 'Test',
    firstName: 'Evhenii',
    gender: null,
    lastName: 'Test',
    phoneNumber: '+299999999',
    street: 'street',
    city: 'city',
    state: 'state',
    zipCode: 'zipCode',
    country: 'country',
  },
  userInvitation: [
    {
      id: '52080e19-c5e0-45bb-af58-edbbe70e78fa',
      created_at: '2022-06-23T06:32:55.588Z',
      email: '12@mail.ru',
      last_sent_at: '2022-06-23T06:32:55.474Z',
      resendable: true,
      status: 'pending',
      user_account_id: '7211ccc2-1fca-4799-b1e1-4005e9fc42be',
    },
  ],
};
