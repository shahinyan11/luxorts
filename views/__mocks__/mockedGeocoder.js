export default [
  {
    streetNumber: '240',
    street: '240 North 7th Avenue',
    city: 'Phoenix',
    state: 'Arizona',
    country: 'United States',
    zipCode: '85007',
    fullAddress: '240 N 7th Ave, Phoenix, AZ 85007, USA',
    longitude: -112.0825682,
    latitude: 33.4507026,
  },
];
