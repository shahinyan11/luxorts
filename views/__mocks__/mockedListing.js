export default {
  id: '0645baf3-945d-4122-8a0b-5ab3e95a9b3e',
  applicationStep: 'accomodation',
  status: 'in_draft',
  userAccount: { id: '6cbf2b68-4a26-41b5-bcd5-161cc37e1420', type: 'hostAccount' },
  listingPropertyType: {
    id: 'ea19f127-bc89-4799-bdbb-346126ac3845',
    typeOfPlace: 'private_place',
    propertyType: {
      id: '51e35121-c27e-46fc-9feb-b51dab2490ee',
      name: 'repudiandae',
      description: 'description repudiandae',
    },
  },
  listingLocation: {
    id: '2b4032ec-a469-4489-b7dc-536d8f9552eb',
    country: 'United States',
    city: 'Phoenix',
    state: 'Arizona',
    street: 'West Maryland Avenue',
    apartmentNumber: '12',
    zipCode: '12345',
    longitude: '-73.97447621784085',
    latitude: '40.764468117826794',
  },
  listingDescription: null,
  listingPricing: {
    pricePerDay: 64,
  },
};
