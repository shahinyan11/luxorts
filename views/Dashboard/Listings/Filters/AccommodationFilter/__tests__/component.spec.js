import { shallow } from 'enzyme';
import { pick } from 'ramda';

import {
  ACCOMMODATION_LISTING_FILTER_KEYS,
  LISTING_FILTERS,
} from 'constants/listing/manageListings';

import { formik } from '__mocks__/formik';

import useContainer from '../hook';
import AccommodationFilter from '../component';

const mockedHookData = {
  visible: true,
  clearButtonVisible: false,
  formik,
  appliedFilters: 0,
  onVisibilityChangeHandler: jest.fn(),
  onApply: jest.fn(),
  onClear: jest.fn(),
};
jest.mock('../hook', () => jest.fn(() => mockedHookData));

describe('AccommodationFilter component', () => {
  const props = {
    filters: pick(ACCOMMODATION_LISTING_FILTER_KEYS, LISTING_FILTERS),
    onFilter: jest.fn(),
  };
  let component = shallow(<AccommodationFilter {...props} />);

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('matches snapshot when clearButtonVisible equals true', () => {
    useContainer.mockReturnValueOnce({
      ...mockedHookData,
      clearButtonVisible: true,
    });

    component = shallow(<AccommodationFilter {...props} />);

    expect(component).toMatchSnapshot();
  });

  it('matches snapshot when appliedFilters equals 1', () => {
    useContainer.mockReturnValueOnce({
      ...mockedHookData,
      appliedFilters: 1,
    });

    component = shallow(<AccommodationFilter {...props} />);

    expect(component).toMatchSnapshot();
  });
});
