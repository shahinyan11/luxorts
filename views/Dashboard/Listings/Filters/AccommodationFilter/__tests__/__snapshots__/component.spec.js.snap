// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccommodationFilter component matches snapshot 1`] = `
<Dropdown
  className="table-head__dropdown-payment"
  mouseEnterDelay={0.15}
  mouseLeaveDelay={0.1}
  onVisibleChange={[MockFunction]}
  overlay={
    <div
      className="table-head__dropdown-content"
    >
      <div
        className="d-flex flex-column mb-0 table-head__content"
      >
        <ForwardRef(InternalForm)
          className="pl-24 pr-24 pt-24 pb-24"
          layout="vertical"
          size="large"
        >
          <Context.Provider
            value={
              Object {
                "dirty": true,
                "handleSubmit": [MockFunction],
                "isValid": true,
                "resetForm": [MockFunction],
                "setFieldTouched": [MockFunction],
                "setFieldValue": [MockFunction],
                "setValues": [MockFunction],
                "validateForm": [MockFunction],
                "values": Object {},
              }
            }
          >
            <InputNumberField
              addonBefore={
                <Memo(FormattedMessage)
                  id="shared.guests"
                />
              }
              formItemClassName="mb-8 width-312 ant-form-item--light"
              max={16}
              min={0}
              name="guests_number-eq"
              withFormItem={true}
            />
            <InputNumberField
              addonBefore={
                <Memo(FormattedMessage)
                  id="shared.bedrooms"
                />
              }
              formItemClassName="mb-8 width-312 ant-form-item--light"
              max={16}
              min={0}
              name="bedrooms_amount-eq"
              withFormItem={true}
            />
            <InputNumberField
              addonBefore={
                <Memo(FormattedMessage)
                  id="shared.beds"
                />
              }
              formItemClassName="mb-8 width-312 ant-form-item--light"
              max={16}
              min={0}
              name="beds_amount-eq"
              withFormItem={true}
            />
            <InputNumberField
              addonBefore={
                <Memo(FormattedMessage)
                  id="shared.baths"
                />
              }
              formItemClassName="mb-0 width-312 ant-form-item--light"
              max={16}
              min={0}
              name="bathrooms_amount-eq"
              withFormItem={true}
            />
          </Context.Provider>
        </ForwardRef(InternalForm)>
      </div>
      <div
        className="table-head__actions pt-24 pb-24 d-flex justify-content-flex-end"
      >
        <GradientButton
          addonAfter={null}
          addonBefore={null}
          className="main-btn--medium min-width-120"
          onClick={[MockFunction]}
          text={
            Object {
              "id": "shared.apply",
            }
          }
          variant="primary"
        />
      </div>
    </div>
  }
  placement="bottomLeft"
  trigger="click"
  visible={true}
>
  <GradientButton
    addonAfter={
      <SvgIcon
        className="icon-right"
        icon="arrow-down"
      />
    }
    addonBefore={null}
    className="main-btn--medium"
    text={
      Object {
        "id": "shared.accommodation",
      }
    }
    variant="filter"
  />
</Dropdown>
`;

exports[`AccommodationFilter component matches snapshot when appliedFilters equals 1 1`] = `
<Dropdown
  className="table-head__dropdown-payment"
  mouseEnterDelay={0.15}
  mouseLeaveDelay={0.1}
  onVisibleChange={[MockFunction]}
  overlay={
    <div
      className="table-head__dropdown-content"
    >
      <div
        className="d-flex flex-column mb-0 table-head__content"
      >
        <ForwardRef(InternalForm)
          className="pl-24 pr-24 pt-24 pb-24"
          layout="vertical"
          size="large"
        >
          <Context.Provider
            value={
              Object {
                "dirty": true,
                "handleSubmit": [MockFunction],
                "isValid": true,
                "resetForm": [MockFunction],
                "setFieldTouched": [MockFunction],
                "setFieldValue": [MockFunction],
                "setValues": [MockFunction],
                "validateForm": [MockFunction],
                "values": Object {},
              }
            }
          >
            <InputNumberField
              addonBefore={
                <Memo(FormattedMessage)
                  id="shared.guests"
                />
              }
              formItemClassName="mb-8 width-312 ant-form-item--light"
              max={16}
              min={0}
              name="guests_number-eq"
              withFormItem={true}
            />
            <InputNumberField
              addonBefore={
                <Memo(FormattedMessage)
                  id="shared.bedrooms"
                />
              }
              formItemClassName="mb-8 width-312 ant-form-item--light"
              max={16}
              min={0}
              name="bedrooms_amount-eq"
              withFormItem={true}
            />
            <InputNumberField
              addonBefore={
                <Memo(FormattedMessage)
                  id="shared.beds"
                />
              }
              formItemClassName="mb-8 width-312 ant-form-item--light"
              max={16}
              min={0}
              name="beds_amount-eq"
              withFormItem={true}
            />
            <InputNumberField
              addonBefore={
                <Memo(FormattedMessage)
                  id="shared.baths"
                />
              }
              formItemClassName="mb-0 width-312 ant-form-item--light"
              max={16}
              min={0}
              name="bathrooms_amount-eq"
              withFormItem={true}
            />
          </Context.Provider>
        </ForwardRef(InternalForm)>
      </div>
      <div
        className="table-head__actions pt-24 pb-24 d-flex justify-content-flex-end"
      >
        <GradientButton
          addonAfter={null}
          addonBefore={null}
          className="main-btn--medium min-width-120"
          onClick={[MockFunction]}
          text={
            Object {
              "id": "shared.apply",
            }
          }
          variant="primary"
        />
      </div>
    </div>
  }
  placement="bottomLeft"
  trigger="click"
  visible={true}
>
  <GradientButton
    addonAfter={
      <SvgIcon
        className="icon-right"
        icon="arrow-down"
      />
    }
    addonBefore={null}
    className="main-btn--medium main-btn--filter-active"
    text={
      Object {
        "id": "shared.accommodationCount",
        "values": Object {
          "count": <span
            className="main-btn__amount"
          >
            1
          </span>,
        },
      }
    }
    variant="filter"
  />
</Dropdown>
`;

exports[`AccommodationFilter component matches snapshot when clearButtonVisible equals true 1`] = `
<Dropdown
  className="table-head__dropdown-payment"
  mouseEnterDelay={0.15}
  mouseLeaveDelay={0.1}
  onVisibleChange={[MockFunction]}
  overlay={
    <div
      className="table-head__dropdown-content"
    >
      <div
        className="d-flex flex-column mb-0 table-head__content"
      >
        <ForwardRef(InternalForm)
          className="pl-24 pr-24 pt-24 pb-24"
          layout="vertical"
          size="large"
        >
          <Context.Provider
            value={
              Object {
                "dirty": true,
                "handleSubmit": [MockFunction],
                "isValid": true,
                "resetForm": [MockFunction],
                "setFieldTouched": [MockFunction],
                "setFieldValue": [MockFunction],
                "setValues": [MockFunction],
                "validateForm": [MockFunction],
                "values": Object {},
              }
            }
          >
            <InputNumberField
              addonBefore={
                <Memo(FormattedMessage)
                  id="shared.guests"
                />
              }
              formItemClassName="mb-8 width-312 ant-form-item--light"
              max={16}
              min={0}
              name="guests_number-eq"
              withFormItem={true}
            />
            <InputNumberField
              addonBefore={
                <Memo(FormattedMessage)
                  id="shared.bedrooms"
                />
              }
              formItemClassName="mb-8 width-312 ant-form-item--light"
              max={16}
              min={0}
              name="bedrooms_amount-eq"
              withFormItem={true}
            />
            <InputNumberField
              addonBefore={
                <Memo(FormattedMessage)
                  id="shared.beds"
                />
              }
              formItemClassName="mb-8 width-312 ant-form-item--light"
              max={16}
              min={0}
              name="beds_amount-eq"
              withFormItem={true}
            />
            <InputNumberField
              addonBefore={
                <Memo(FormattedMessage)
                  id="shared.baths"
                />
              }
              formItemClassName="mb-0 width-312 ant-form-item--light"
              max={16}
              min={0}
              name="bathrooms_amount-eq"
              withFormItem={true}
            />
          </Context.Provider>
        </ForwardRef(InternalForm)>
      </div>
      <div
        className="table-head__actions pt-24 pb-24 d-flex justify-content-flex-end"
      >
        <GradientButton
          addonAfter={null}
          addonBefore={null}
          className="main-btn--medium mr-16"
          onClick={[MockFunction]}
          text={
            Object {
              "id": "shared.clear",
            }
          }
          variant="tertiary"
        />
        <GradientButton
          addonAfter={null}
          addonBefore={null}
          className="main-btn--medium min-width-120"
          onClick={[MockFunction]}
          text={
            Object {
              "id": "shared.apply",
            }
          }
          variant="primary"
        />
      </div>
    </div>
  }
  placement="bottomLeft"
  trigger="click"
  visible={true}
>
  <GradientButton
    addonAfter={
      <SvgIcon
        className="icon-right"
        icon="arrow-down"
      />
    }
    addonBefore={null}
    className="main-btn--medium"
    text={
      Object {
        "id": "shared.accommodation",
      }
    }
    variant="filter"
  />
</Dropdown>
`;
